dev/benchmarks/complex_layout/lib/main.dart

```diff
@@ -0,0 +1,559 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import 'drawer.dart';
+
+void main() {
+  runApp(
+    new FlutterDemoApp()
+  );
+}
+
+class FlutterDemoApp extends StatefulWidget {
+  @override
+  FlutterDemoAppState createState() => new FlutterDemoAppState();
+
+  static FlutterDemoAppState of(BuildContext context) => context.ancestorStateOfType(const TypeMatcher<FlutterDemoAppState>());
+}
+
+class FlutterDemoAppState extends State<FlutterDemoApp> {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      theme: lightTheme ? new ThemeData.light() : new ThemeData.dark(),
+      title: 'Advanced Layout',
+      routes: <String, WidgetBuilder>{
+        '/': (BuildContext context) => new FlutterDemo(),
+      }
+    );
+  }
+
+  bool _lightTheme = true;
+  bool get lightTheme => _lightTheme;
+  void set lightTheme(bool value) {
+    setState(() {
+      _lightTheme = value;
+    });
+  }
+}
+
+class FlutterDemo extends StatefulWidget {
```

Comment:
Maybe rename these classes from "flutter demo" to something about "complex layout" ?